### PR TITLE
[Threading] Always round up when converting std::chrono::durations.

### PR DESCRIPTION
**Pull request by al45tair**
We were inadvertently rounding down, which was fine almost 100% of the time, but on Windows where the timer resolution is only 1ms, it made the test _very slightly_ flaky because we were asking to wait for 9ms rather than 10ms, then checking we'd waited at least 10ms.

We should explicitly round up, for that reason.

rdar://100236038

### CODE REVIEW

**Comment by al45tair** [comment]
@swift-ci Please test

**Comment by al45tair** [comment]
Hmmm. Seems this wasn't the only problem on Windows making the test flaky. I wonder if the server applied a time adjustment during the `SleepConditionVariableSRW()` call. Maybe the thing to do is to change the test to allow things to return 10% "sooner" than it thinks they should?

**Comment by al45tair** [comment]
No, I don't think the time adjustment explanation is the right one. That _shouldn't_ really be happening because we're using `steady_clock`. I think it might be the odd behaviour of `Sleep` functions on Windows, where (apparently) if you try to wait for less than a system tick (by default 1/64s), they can return _early_.

Let's try increasing the system tick rate during the test. I'll run the tests a few times to make sure it works.

**Comment by al45tair** [comment]
@swift-ci Please test Windows platform

**Review by mikeash — APPROVED** [review]
LGTM, once it works.

**Comment by al45tair** [comment]
@swift-ci Please test Windows platform

**Comment by al45tair** [comment]
@swift-ci Please test Windows platform

File: include/swift/Threading/Impl/chrono.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+//===--- chrono.h - Utility functions for duration/time_point - -*- C++ -*-===//
```
Comment (compnerd) [inline include/swift/Threading/Impl/chrono.h:1]: Does it make sense to rename this header to `chrono_utils.h` to match the namespace and to differentiate it from the C++ header `chrono`?

**Review by compnerd — APPROVED** [review]

**Comment by al45tair** [comment]
@swift-ci Please test Windows platform

**Comment by al45tair** [comment]
@swift-ci Please test Windows platform

**Comment by al45tair** [comment]
Woo! That seems to have worked.

**Comment by al45tair** [comment]
@swift-ci please smoke test

**Pull request merged**